lib/forms/CoBrALab-Mouse-Motor-Task-Form/index.ts

```diff
@@ -0,0 +1,186 @@
+/* eslint-disable perfectionist/sort-objects */
+
+const { defineInstrument } = await import('/runtime/v1/@opendatacapture/runtime-core/index.js');
+const { z } = await import('/runtime/v1/zod@3.23.6/index.js');
+
+
+function createDependentField<T>(field: T, fn: (motorTask: string) => boolean) {
+  return {
+    kind: 'dynamic' as const,
+    deps: ['motorTask'] as const,
+    render: (data: { motorTask: string }) => {
+      if (fn(data.motorTask)) {
+        return field;
+      }
+      return null;
+    }
+  };
+}
+
+export default defineInstrument({
+  kind: 'FORM',
+  language: 'en',
+  tags: ['Rotorod', 'Motor tasks', 'Wire Hang', 'Pole test'],
+  internal: {
+    edition: 1,
+    name: 'MOUSE_MOTOR_TASK_FORM'
+  },
+  content: {
+    roomNumber: {
+      kind: "string",
+      variant: "input",
+      label: "Room number"
+    },
+    motorTask: {
+      kind: "string",
+      variant: "select",
+      label: "Motor task",
+      options: {
+        "Rotorod": "Rotorod",
+        "Wire hang": "Wire hang",
+        "Pole test": "Pole test"
+      }
+    },
+    rotorodMiceNumber: createDependentField({
+      kind: "number",
+      variant: "slider",
+      label: "Number of mice",
+      max: 5,
+      min: 1
+    },(type) => type === "Rotorod"),
+
+    rotorodDuration:createDependentField({
+      kind: "number",
+      variant: "input",
+      label: "Duration before falling (seconds)"
+    },(type) => type === "Rotorod"),
+
+    rotorodSlotPosition: createDependentField({
+      kind: "string",
+      variant: "select",
+      label: "Rotorod position",
+      options: {
+        "Left most": "Left most",
+        "Middle left": "Middle left",
+        "Middle right": "Middle right",
+        "Right most": "Right most"
+      }
+    }, (type) => type === "Rotorod"),
+    rotorodWirehangFailure: createDependentField({
+      kind: "boolean",
+      variant: "radio",
+      label: "Mouse failed session"
+    }, (type) => type === "Rotorod" || type === "Wire hang" ),
+    
+    wirehangDuration:createDependentField({
+      kind: "number",
+      variant: "input",
+      label: "Duration before falling (seconds)"
+    },(type) => type === "Wire hang"),
+
+    wirehangPutbacks: createDependentField({
+      kind: "number",
+      variant: "input",
+      label: "Times put back on wire"
```

Comment:
If this happens multiple times, what does the time to fall mean now?

Reply:
True this can lead to confusion, I can switch it to longest duration before falling.